### DISCUSSION TITLE
how to create a peptide bond from two monomer

### DISCUSSION THREAD

**Discussion by hahadg** [rdkit/rdkit#8057] [Q&A] [answered]
Hi everyone,

I want to create the peptide bond from two amino acids (natural or non-natural) using RDKIT. My attempt is as follows:
```
from rdkit import Chem
from rdkit.Chem import AllChem

rxn = AllChem.ReactionFromSmarts('[OD1][C:1](=[O:2])[C:3]([N!H0:4])([*]).[OD1][C:5](=[O:6])[C:7]([N!H0:8])([*])>>[OD1][C:1](=[O:2])[C:3]([*])[N!H0:4][C:5](=[O:6])[C:7]([N:8])([*])')
reactants = ((Chem.MolFromSmiles('NC(CS)C(O)=O'), Chem.MolFromSmiles('NC(CC1=CNC2=C1C=CC=C2)C(O)=O')))
products = rxn.RunReactants(reactants)
print(products)

for mol in products:
    print(Chem.MolToSmiles(mol[0]))

```
And the output is as follows:
![image](https://github.com/user-attachments/assets/61531319-54a7-4954-8348-66dc33a2d668)
![image](https://github.com/user-attachments/assets/4619b1e8-0235-4a4b-bc34-37ceecdb9492)
**Q1** : What's wrong with my reaction SMARTS?

**Q2**: Actually, I have tried another reaction SMARTS (as follows), but **when there are non-natural amino acids which have more than one -NH2 or -COOH, there will be more than one reaction products. I want to give priority to the reaction on the Cα**.
`rxn = AllChem.ReactionFromSmarts('[C:1](=[O:2])-[OD1].[N!H0:3]>>[C:1](=[O:2])[N:3]')`

Any suggestions will be appreciated!

**Comment by ptosco** [comment, marked as answer]
@Nicole-DH It is best to only map atoms which are involved in the reaction, as in your 2nd attempt, where however you are mapping the carbonyl oxygen (which is also present in the product) rather than the OH oxygen.
To avoid the formation of undesired products you may specify the atomic environment of reactive atoms using recursive SMARTS.

Here you are coupling an amine to a carboxylic acid, so you can simplify your reaction to:

R<sub>1</sub>-NH-R<sub>2</sub> + R<sub>3</sub>-C(=O)-OH  ---> R<sub>1</sub>-N(-R<sub>2</sub>)-C(=O)-R<sub>3</sub> + H<sub>2</sub>O

There are only three atoms taking active part to the reaction:
1. the carbonyl carbon on the LHS amino acid
2. the OH oxygen on the LHS amino acid
3. the amine nitrogen on the RHS amino acid

A new bond forms between (1) and (3), while the existing bond between (1) and (2) breaks and (2) is eliminated.
This assumes that the OH oxygen is consistently eliminated as water from the tetrahedral intermediate and the carbonyl oxygen consistently retained. This assumption is acceptable as long as your oxygens are not isotopically labeled. If they were, you should modify your reaction in order to get both products.

I would write this reaction using recursive SMARTS, mapping only the reactive atoms (i.e., the first atom in the recursive SMARTS patterns) and specifying the atomic environment of the reactive atoms in the recursive SMARTS pattern on the right of the first reactive atom, to the extent needed to avoid the formation of undesired products and to eventually prevent reactants which do not match the AA pattern from reacting.

```python
from rdkit import Chem
from rdkit.Chem import AllChem
from rdkit.Chem.Draw import MolsToGridImage

reactants = ((Chem.MolFromSmiles('NC(CS)C(O)=O'), Chem.MolFromSmiles('NC(CC1=CNC2=C1C=CC=C2)C(O)=O')))
MolsToGridImage(reactants)
```
![image](https://github.com/user-attachments/assets/0a8ea421-ef47-4997-a197-a0733ef9656a)

```python
rxn = AllChem.ReactionFromSmarts(
    '[$([NX3!H0][CX4][CX3](=[OX1])[OD1]):1].[$([CX3](=[OX1])[CX4][NX3!H0]):2]-[OD1:3]>>'
    '[NX3:1]-[CX3:2].[OD0:3]'
)
products = rxn.RunReactants(reactants)
print(products)
((<rdkit.Chem.rdchem.Mol object at 0x7fdf2b54fbc0>, <rdkit.Chem.rdchem.Mol object at 0x7fdf2b54f920>),)

for mol in products:
    Chem.SanitizeMol(mol[0])
    print(Chem.MolToSmiles(mol[0]))
NC(Cc1c[nH]c2ccccc12)C(=O)NC(CS)C(=O)O

products[0][0]
```
![image](https://github.com/user-attachments/assets/ce077902-10c4-4155-b4ad-b1844c9e28f8)
Should you find out that the above pattern is not specific enough to prevent the formation of undesired products, you can make it as specific as it needs to be by adjusting the recursive SMARTS patterns.

**Reply by hahadg** [reply]
Hi Paolo,

Thanks a lot for your quick response. It perfectly solved my problem.
Following your advise, I implement a recursive SMARTS for coupling an amine to a carboxylic acid between the Cβ and the Cγ residues. The results are exactly what I needed.

```
rxn = AllChem.ReactionFromSmarts(
    '[$([NX3!H0][CX4][CX4][CX3](=[OX1])[OD1]):1].[$([CX3](=[OX1])[CX4][CX4][CX4][NX3!H0]):2]-[OD1:3]>>'
    '[NX3:1]-[CX3:2].[OD0:3]'
)

reactants = ((Chem.MolFromSmiles('N[C@@H]1CCC[C@H]1C(O)=O'), Chem.MolFromSmiles('O=C(C1CCNCC1)O')))
products = rxn.RunReactants(reactants)
print(products)

for mol in products:
    print(Chem.MolToSmiles(mol[0]))
```

`O=C(N[C@@H]1CCC[C@H]1C(=O)O)C1CCNCC1`

![image](https://github.com/user-attachments/assets/b3445c21-e2db-4e3c-a8c0-cca87a519bc5)

By the way, I would also like to ascertain the specific type of input amino acid, such as α-, β-, or γ-amino acids, using the `HasSubstructMatch()` function. Below is an implementation example.

```
from rdkit import Chem
sub = Chem.MolFromSmarts('[$([NX3!H0][CX4][CX4][CX3](=[OX1])[OD1]):1]')     # β-amino acids
mol = Chem.MolFromSmiles('N[C@@H]1CCC[C@H]1C(O)=O')
mol.HasSubstructMatch(sub)
```

`True`

It appears to be feasible. Do you have any suggestions?

Thanks again for your help！
